### PR TITLE
chore: aries-js-worker - didexchange edgeagent/router

### LINKED ISSUE #1330: aries-js-worker: add bdd test: didexchange b/w edge agent  and router
part of #1238

### PR DESCRIPTION
**Pull request by llorllale**
closes #1330

* BDD test for did-exchange edgeagent <-> router
* fixes for properly propagating errors wasm -> JS when startup fails (eg. invalid startup options) and when `didexchange.acceptInvitation` fails
* `didexchange.acceptInvitation/queryConnectionByID` now accept a text argument and transform it to a properly built object that the wasm expects
* better test reporting to the terminal

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

File: cmd/aries-js-worker/src/aries.js [hunk 1]
```diff
@@ -183,11 +181,16 @@ const Aries = function(opts) {
...
                 return invoke(aw, pending,  this.pkgname, "CreateInvitation", text, "timeout while creating invitation")
             },
-            receiveInvitation: async function (text) {
-                return invoke(aw, pending,  this.pkgname, "ReceiveInvitation", text, "timeout while receiving invitation")
+            receiveInvitation: async function (invitation) {
+                return invoke(aw, pending,  this.pkgname, "ReceiveInvitation", invitation, "timeout while receiving invitation")
             },
-            acceptInvitation: async function (text) {
-                return invoke(aw, pending,  this.pkgname, "AcceptInvitation", text, "timeout while accepting invitation")
+            acceptInvitation: async function (connectionID) {
```
Comment (rolsonquadras) [inline cmd/aries-js-worker/src/aries.js]: The accept invitation API takes optional public DID as well (https://github.com/hyperledger/aries-framework-go/blob/master/pkg/controller/command/didexchange/models.go#L86) along with connection ID.

Comment (llorllale) [reply cmd/aries-js-worker/src/aries.js]: renamed to "id"

Comment (llorllale) [reply cmd/aries-js-worker/src/aries.js]: ok: renamed back to `connectionID` and added an optional `publicDID` arg

File: cmd/aries-js-worker/src/aries.js [hunk 2]
```diff
@@ -183,11 +181,16 @@ const Aries = function(opts) {
...
                 return invoke(aw, pending,  this.pkgname, "CreateInvitation", text, "timeout while creating invitation")
             },
-            receiveInvitation: async function (text) {
-                return invoke(aw, pending,  this.pkgname, "ReceiveInvitation", text, "timeout while receiving invitation")
+            receiveInvitation: async function (invitation) {
+                return invoke(aw, pending,  this.pkgname, "ReceiveInvitation", invitation, "timeout while receiving invitation")
             },
-            acceptInvitation: async function (text) {
-                return invoke(aw, pending,  this.pkgname, "AcceptInvitation", text, "timeout while accepting invitation")
+            acceptInvitation: async function (connectionID, publicDID="") {
```
Comment (sudeshrshetty) [inline cmd/aries-js-worker/src/aries.js]: As per my opinion, we should avoid using strict function args, because every time there is an update in command controller models we may have to come back here to make corresponding updates.

File: test/aries-js-worker/test/didexchange/edgeagent_router.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                aries.didexchange.acceptInvitation(connection.connection_id)
+            } catch (err) {
+                done(err)
+            }
+            done()
+        }, ["connections"])
+        aries.didexchange.receiveInvitation(invitation)
+    })
+
+    it("Edge Agent validates that the connection's state is 'completed'", function(done) {
```
Comment (rolsonquadras) [inline test/aries-js-worker/test/didexchange/edgeagent_router.js]: Are we going to have the .feature file for this step/scenario ?

Comment (llorllale) [reply test/aries-js-worker/test/didexchange/edgeagent_router.js]: See my findings so far under the second point here in #1238

Comment (llorllale) [reply test/aries-js-worker/test/didexchange/edgeagent_router.js]: So short answer is - I couldn't make it work.

Comment (rolsonquadras) [reply test/aries-js-worker/test/didexchange/edgeagent_router.js]: ok. so, for now we wont be able to reuse the steps.
